nexus/reconfigurator/execution/src/lib.rs

```diff
@@ -90,6 +90,21 @@ where
     omicron_zones::deploy_zones(&opctx, &sleds_by_id, &blueprint.omicron_zones)
         .await?;
 
+    // After deploying omicron zones, we may need to refresh OPTE service
+    // firewall rules. This is an idempotent operation, so we don't attempt
+    // to optimize out calling it in unnecessary cases, although we expect
+    // _most_ cases this is not needed.
+    nexus_networking::plumb_service_firewall_rules(
+        datastore,
+        &opctx,
+        &[],
+        &opctx,
+        &opctx.log,
+    )
+    .await
+    .context("failed to plumb service firewall rules to sleds")
+    .map_err(|err| vec![err])?;
+
```

Comment:
Just to triple-check -- this is the real meat of this PR, and the rest is more-or-less refactoring?

Reply:
Correct